integrations/kube-agent-updater/pkg/version/basichttp.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/rogpeppe/go-internal/semver"
+
+	"github.com/gravitational/teleport/integrations/kube-agent-updater/pkg/cache"
+)
+
+const (
+	versionPath   = "version"
+	httpTimeout   = 10 * time.Second
+	cacheDuration = time.Minute
+)
+
+// basicHTTPVersionClient retrieves the version from an HTTP endpoint
+// it should not be invoked immediately and must be wrapped in a cache layer in
+// order to avoid spamming the version server in case of reconciliation errors.
+// use BasicHTTPVersionGetter if you need to get a version.
+type basicHTTPVersionClient struct {
+	baseURL *url.URL
+	client  *http.Client
+}
+
+// Get sends an HTTP GET request and returns the version prefixed by "v".
+// It expects the endpoint to be unauthenticated, return 200 and the response
+// body to contain a valid semver tag without the "v".
+func (b *basicHTTPVersionClient) Get(ctx context.Context) (string, error) {
+	versionURL := b.baseURL.JoinPath(versionPath)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, versionURL.String(), nil)
```

Comment:
can we reuse the same code as above?

Reply:
Do you mean the maintenance getter code or the ping version getter one?

Reply:
the maintenance code but roman also pointed that out